pkg/alertmanager/alertmanager_metrics_test.go

```diff
@@ -257,6 +258,474 @@ func TestAlertmanagerMetricsStore(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestAlertmanagerMetricsRemoval(t *testing.T) {
+	mainReg := prometheus.NewPedanticRegistry()
+
+	alertmanagerMetrics := newAlertmanagerMetrics()
+	mainReg.MustRegister(alertmanagerMetrics)
+	alertmanagerMetrics.addUserRegistry("user1", populateAlertmanager(1))
+	alertmanagerMetrics.addUserRegistry("user2", populateAlertmanager(10))
+	alertmanagerMetrics.addUserRegistry("user3", populateAlertmanager(100))
+
+	// Assertion before removal.
+	err := testutil.GatherAndCompare(mainReg, bytes.NewBufferString(`
```

Comment:
Nit: It's difficult to see what's being tested here. Couple of metrics (some gauge, some per-user, some counter) would be enough, instead of all :)

Reply:
Let me add a more in-depth comment, I tend to prefer these kinds of holistic tests as they're susceptible to small changes. With that, it makes it clear to see when something does change which tends to be helpful with regressions.

Reply:
Done!